tests/test_composite.py

```diff
@@ -39,3 +39,12 @@ def children(self):
         sampler = Dummy()
         with self.assertRaises(RuntimeError):
             sampler.child
+
+
+class TestInnermostChild(unittest.TestCase):
+
+    def test_composed_sampler(self):
+        innermost_child = dimod.ExactSolver()
+        sampler = dimod.ClipComposite(dimod.ScaleComposite(innermost_child))
```

Comment:
I would use `TrackingComposite` and `StructureComposite` since those are staying in dimod through #1127 so the tests stay valid.